pkg/compact/compact.go

```diff
@@ -470,6 +481,98 @@ func (cg *Group) Resolution() int64 {
 	return cg.resolution
 }
 
+// metrics related to planning and compaction progress
+type ProgressMetrics struct {
+	NumberOfIterations    *prometheus.GaugeVec
+	NumberOfBlocksToMerge *prometheus.GaugeVec
+	// figure out where to add groupKey labels to this
+}
+
+// should return the results/metrics of the planning simulation
+type PlanSim interface {
+	ProgressCalculate(ctx context.Context, groups []*Group) error
+}
+
+type DefaultPlanSim struct {
+	planner Planner
+	ProgressMetrics
+}
+
+func NewDefaultPlanSim(reg prometheus.Registerer, logger log.Logger) *DefaultPlanSim {
```

Comment:
Why not just pass the planner here? This way we can reuse the TSDB planner we already initialized before

Reply:
Have you updated that in your latest commit? I don't see u are using the tsdb planner